Fix compilation error for Linux 6.3

fewletter opened the pull request.

fewletter committed:
Fix compile error Linux v6.3+

jserv reviewed (changes requested):
Read https://cbea.ms/git-commit/ carefully and improve git commit message. You shall summarize the changes of kernel internal functions.

jserv reviewed (changes requested):
Read https://cbea.ms/git-commit/ carefully and improve git commit message. You shall summarize the changes of kernel internal functions.

jserv reviewed (changes requested):
Read https://cbea.ms/git-commit/ carefully and improve git commit message. You shall summarize the changes of kernel internal functions.

jserv commented on simplefs.h:
```diff
@@ -27,6 +27,7 @@
 #include <linux/version.h>
 
 #define USER_NS_REQUIRED() LINUX_VERSION_CODE >= KERNEL_VERSION(5,12,0)
+#define USER_NS_REQUIRED_6_3() LINUX_VERSION_CODE >= KERNEL_VERSION(6,3,0)
```
`USER_NS_REQUIRED_6_3` is not an appropriate name. Instead, check the kernel commit message and summarize the reason why kernel developers changed the usage. Finally, reflect the considerations here.

fewletter committed:
Adjust the name of definition by using the function parameter

USER_NS_REQUIRED_6_3 is not accurate to illustrate the definition. Therefore, the definition name changes into MNT_IDMAP_REQUIRED to explain function more clearly.

jserv commented:
Thank @fewletter for contributing! I did amend the git commit messages. Next time, write down more precise ones.
